packages/story-editor/src/components/videoTrim/useVideoNode.js

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useEffect, useState, useCallback } from '@web-stories-wp/react';
+
+function useVideoNode() {
+  const [currentTime, setCurrentTime] = useState(0);
+  const [startOffset, rawSetStartOffset] = useState(0);
+  const [maxOffset, setMaxOffset] = useState(Math.POSITIVE_INFINITY);
+  const [endOffset, rawSetEndOffset] = useState(Math.POSITIVE_INFINITY);
+  const [videoNode, setVideoNode] = useState(null);
+
+  useEffect(() => {
+    if (!videoNode) {
+      return undefined;
+    }
+
+    function onLoadedMetadata(evt) {
+      const duration = Math.floor(evt.target.duration * 1000);
+      rawSetStartOffset(0);
+      setCurrentTime(0);
+      rawSetEndOffset(duration);
+      setMaxOffset(duration);
+    }
+    function onTimeUpdate(evt) {
+      const currentOffset = Math.floor(evt.target.currentTime * 1000);
+      setCurrentTime(currentOffset);
+      if (currentOffset >= endOffset) {
+        setTimeout(() => {
+          videoNode.currentTime = startOffset / 1000;
```

Comment:
This converting back and forth between seconds and milliseconds is a bit annoying, but I feel it's cleaner to have our API in millis. Is it, though?